File: 485-max-consecutive-ones.js
var findMaxConsecutiveOnes = function(nums) {
    let max = 0;
    let count = 0;
    for (let i = 0; i < nums.length; i++) {
        count++;
        if (nums[i + 1] !== nums[i]) {
            if (count > max) max = count;
            count = 0;
        }
    }
    return (count > max) ? count : max;
};

File: 485-max-consecutive-ones.rb
def find_max_consecutive_ones(nums)
    max = 0;
    count = 0;
    nums.each_with_index do |el, idx|
        count += 1
        if nums[idx + 1] != el
            max = count if count > max
            count = 0
        end
    end
    count > max ? count : max
end
